aspnetcore/fundamentals/startup.md

```diff
@@ -65,7 +65,57 @@ Each `Use` extension method adds a [middleware](xref:fundamentals/middleware) co
 
 For more information about how to use `IApplicationBuilder`, see [Middleware](xref:fundamentals/middleware).
 
-Additional services, like `IHostingEnvironment` and `ILoggerFactory` may also be specified in the method signature, in which case these services will be [injected](dependency-injection.md) if they are available. 
+Additional services, like `IHostingEnvironment` and `ILoggerFactory` may also be specified in the method signature, in which case these services will be [injected](dependency-injection.md) if they are available.
+
+## Hosting startup light up feature
```

Comment:
This is missing the main point of the feature which is to inject code from outside of the application. Lighting up code in your own assembly was incidental.

This lets you opt into features like azure site extensions for AI without redeploying your application. They're activated by an environment variable that Hosting reads (and an second variable for additional deps in the .NET Core runtime). The binary also needs to be in a loadable location such as the runtime store, the bin directory, the GAC, etc..

Reply:
I see. I thought it was just an outside-the-assembly thing such that showing it with the attribute would be enough. I'll try a new sample that demos it the way that you describe.